docs/aws/audit/redshiftmonitoring/rules/redshift_reserved_node_recent_purchase_remediation.mdx

```diff
@@ -1,6 +1,244 @@
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard. You can do this by typing "Redshift" into the search bar at the top of the console, then selecting "Amazon Redshift" from the dropdown menu.
+3. Once in the Redshift dashboard, click on "Reserved Nodes" in the left-hand navigation pane. This will bring up a list of all your reserved nodes.
+4. Review the list of reserved nodes for any recent purchases. Look at the "Start Date" and "Duration" columns to determine when the node was purchased and for how long it is reserved. If there are any nodes that were recently purchased, they should be reviewed to ensure they are necessary and being used efficiently.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure to configure it with the necessary access keys and region.
+
+2. Once the AWS CLI is set up, you can use the `describe-reserved-nodes` command to list all the Redshift reserved nodes. The command is as follows:
+
+   ```
+   aws redshift describe-reserved-nodes
+   ```
+
+3. This command will return a JSON output with details about all the reserved nodes. You need to review the `StartTime` field for each reserved node. This field indicates when the reserved node was purchased.
+
+4. If the `StartTime` is within the last few days, it means that the reserved node was recently purchased and should be reviewed. You can use a script to automate this process. Here is a simple Python script that uses the `boto3` library to check for recent purchases:
+
+   ```python
+   import boto3
+   from datetime import datetime, timedelta
+
+   # Create a Redshift client
+   redshift = boto3.client('redshift')
+
+   # Get the current date and time
+   now = datetime.now()
+
+   # Get all the reserved nodes
+   response = redshift.describe_reserved_nodes()
+
+   # Check each reserved node
+   for node in response['ReservedNodes']:
+       # Get the start time of the reserved node
+       start_time = node['StartTime']
+
+       # Check if the reserved node was purchased within the last few days
+       if now - start_time < timedelta(days=3):
+           print(f"Reserved node {node['ReservedNodeId']} was recently purchased and should be reviewed.")
+   ```
+
+   This script will print out the IDs of all reserved nodes that were purchased within the last three days.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your environment. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>